src/main/java/com/twilio/oai/api/CsharpApiResourceBuilder.java

```diff
@@ -78,14 +89,31 @@ public CsharpApiResources build() {
     @Override
     public ApiResourceBuilder updateOperations(Resolver<CodegenParameter> codegenParameterIResolver) { // CsharpParameterResolver
         super.updateOperations(codegenParameterIResolver);
+        processAuthMethods(this.codegenOperationList);
         this.codegenOperationList.forEach(co -> {
             co.headerParams.forEach(e -> codegenParameterIResolver.resolve(e, this));
             populateRequestBodyArgument(co);
             resolveIngressModel(co);
         });
-
         return this;
     }
+
+    public void processAuthMethods(List<CodegenOperation> opList) {
+        if(opList != null){
+            List<CodegenSecurity> authMethods = opList.get(0).authMethods;
+            if(authMethods != null){
+                for(CodegenSecurity c : authMethods){
+                    if(c.isOAuth == true){
+                        this.authMethod = "BearerToken";
```

Comment:
public apis will have basic and token based authentication, We don't want to re-write this code again for OAuth for public APIs.

Reply:
If Oauth is added to public(Non Orgs) APIs, This code will break existing public APIs